force-app/main/releaseGating/classes/ReleaseGateAPIService.cls

```diff
@@ -0,0 +1,230 @@
+            (String) result,
+            GetReleaseGateActionResultModel.class
+        );
+        this.verifyCallableResult(resultModel, GET_ACTION);
+        return resultModel;
+    }
+
+    /**
+     * @description Calls the specified callable action
+     * @param  action     action name to call
+     * @param  parameters action parameters
+     * @return            Callable return result
+     */
+    protected virtual Object makeCall(String action, Map<String, Object> parameters) {
+        Callable callableInstance = this.locateReleaseGateCallable();
+        UTIL_Debug.debug(
+            String.format(
+                'Calling {0}.{1} action, version {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, parameters.get(VERSION_PARAMETER) }
+            )
+        );
+        Object result;
+        try {
+            result = callableInstance.call(action, parameters);
+        } catch (Exception ex) {
+            String errorMessage = String.format(
+                'An unhandled exception occurred during execution of {0}.{1}: {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, ex.getMessage() }
+            );
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, errorMessage);
+            throw new ReleaseGateCallableException(
+                'An unhandled exception occurred during execution: ' + errorMessage,
+                ex,
+                this.releaseGateRegistry,
+                null
+            );
+        }
+        UTIL_Debug.debug(
+            String.format(
+                'Finished {0}.{1}, got the following response: \r\n {2}',
+                new List<Object>{ this.releaseGateRegistry.name, action, result }
+            )
+        );
+        return result;
+    }
+
+    /**
+     * @description Verify if the result is successful, throw ReleaseGateCallableException if not
+     * @param  resultModel Callable action result
+     * @param  action      Callable action name
+     */
+    private void verifyCallableResult(CallableResult resultModel, String action) {
+        if (!resultModel.success) {
+            String errorMessage = resultModel.error != null
+                ? String.valueOf(resultModel.error.code) + ': ' + resultModel.error.message
+                : 'Unspecified error';
+            errorMessage += String.format(
+                ', action: {0}, class {1}, namespace {2}',
+                new List<Object>{ action, this.releaseGateRegistry.classname, this.releaseGateRegistry.namespace }
+            );
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, errorMessage);
+            throw new ReleaseGateCallableException(errorMessage, null, this.releaseGateRegistry, resultModel);
+        }
+    }
+
+    /**
+     * ReleaseGating product/class not found custom exception class
+     */
+    public class ReleaseGateProductNotFoundException extends Exception {
```

Comment:
Not necessary at this moment, but please make a constructor test for this in the error handling PR.

Reply:
Noted, will add test in the next user story